app/generate_protos.sh

```diff
@@ -1,26 +1,27 @@
 #!/bin/sh
+set -e
 
 # generate API protos and grpc stuff
-find pb -name '*.proto' | protoc -I. \
+cd proto && find . -name '*.proto' | protoc \
   --plugin=protoc-gen-grpc_python=$(which grpc_python_plugin) \
   --include_imports --include_source_info \
   \
-  --descriptor_set_out proxy/protos.pb \
+  --descriptor_set_out ../proxy/protos.pb \
   \
-  --python_out=backend/src \
-  --grpc_python_out=backend/src \
+  --python_out=../backend/src/proto \
+  --grpc_python_out=../backend/src/proto \
   \
-  --python_out=media/src \
-  --grpc_python_out=media/src \
+  --python_out=../media/src/proto \
+  --grpc_python_out=../media/src/proto \
   \
-  --js_out="import_style=commonjs,binary:frontend/src" \
-  --grpc-web_out="import_style=commonjs+dts,mode=grpcweb:frontend/src" \
+  --js_out="import_style=commonjs,binary:../frontend/src/proto" \
+  --grpc-web_out="import_style=commonjs+dts,mode=grpcweb:../frontend/src/proto" \
   \
-  $(xargs)
+  $(xargs) && cd ..
```

Comment:
Split into two commands, one for Python and one for Javascript. Keep the old style for Python.